packages/core/src/internal/validation/futures/validateNamedEncodeFunctionCall.ts

```diff
@@ -0,0 +1,66 @@
+import { IgnitionError } from "../../../errors";
+import {
+  isArtifactType,
+  isModuleParameterRuntimeValue,
+} from "../../../type-guards";
+import { ArtifactResolver } from "../../../types/artifact";
+import { DeploymentParameters } from "../../../types/deploy";
+import { EncodeFunctionCallFuture } from "../../../types/module";
+import { ERRORS } from "../../errors-list";
+import { validateArtifactFunction } from "../../execution/abi";
+import { retrieveNestedRuntimeValues } from "../utils";
+
+export async function validateNamedEncodeFunctionCall(
```

Comment:
I was going to query why this was not just `validateEncodeFunctionCall`, but it is following the pattern of `validateNamedStaticCall`. That raises the further question of whether we intended this pattern for the calls or whether it is just an accident of transcription.
I don't think we solve this here by the way, but do we need another PR for these?